hedera-mirror-grpc/src/main/java/com/hedera/mirror/grpc/listener/TopicListenerImpl.java

```
@@ -0,0 +1,97 @@
+
+import com.hedera.mirror.grpc.domain.TopicMessage;
+
+import com.hedera.mirror.grpc.domain.TopicMessageFilter;
+
+import io.r2dbc.pool.ConnectionPool;
+import io.r2dbc.postgresql.PostgresqlConnectionFactory;
+import io.r2dbc.spi.ConnectionFactory;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Flux;
+import io.r2dbc.postgresql.api.Notification;
+import reactor.core.publisher.Mono;
+
+@Component
+@Log4j2
+public class TopicListenerImpl implements TopicListener {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+            .setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
+
+    private final Flux<TopicMessage> topicMessages;
+
+    public TopicListenerImpl(ConnectionFactory connectionFactory) {
+        ConnectionPool connectionPool = (ConnectionPool) connectionFactory;
+        PostgresqlConnectionFactory postgressqlConnectionFactory = (PostgresqlConnectionFactory) connectionPool.unwrap();
+        topicMessages = postgressqlConnectionFactory.create().flatMapMany(it -> {
+            return it.createStatement("LISTEN topic_message")
+                    .execute()
+                    .thenMany(it.getNotifications())
+                    .map(this::toTopicMessage);
+        }).share();
+    }
+
+    public Flux<TopicMessage> listen(TopicMessageFilter filter) {
+        return topicMessages.filterWhen(t -> asyncFilterMessage(t, filter));
+    }
+
+    private TopicMessage toTopicMessage(Notification notification) {
+        String parameter = notification.getParameter();
+        TopicMessage topicMessage = null;
+        try {
+            topicMessage = OBJECT_MAPPER.readValue(parameter, TopicMessage.class);
+            log.debug("Listener: {}", topicMessage.getSequenceNumber());
+        }
+        catch (Exception ex) {
+            log.error("OBJECT_MAPPER failed to parse TopicMessage type from string: {}", parameter, ex);
```

Don't ignore errors, re-throw as RuntimeException with logging to halt execution.